src/iexamine.cpp

```diff
@@ -557,6 +557,9 @@ class atm_menu
 
             add_msg( m_info, "amount: %d", amount );
             u.use_charges( itype_cash_card, amount );
+            if( u.cash < 100000000 && ( u.cash + amount ) >= 100000000 ) {
+                g->events().send<event_type::becomes_millionaire>( u.getID() );
+            }
```

Comment:
The events you've added are very niche.  I feel like this one should e.g. be a generic 'deposit' event, which includes the amount deposited and the new balance.  I think deposits are rare enough that shouldn't overwhelm the log.

Reply:
What you suggest here is future-proofing, even though I can't imagine any specific use for it offhandedly, so I'll do it.